File: HackerEarth/missiles.go
package main

import "fmt"

func main() {
	var n, m, p, a1, a2, b1,b2 int
	fmt.Scan(&n)
	a := [n][n]int{}
	fmt.Scan(&m)
	for i := 0; i < n; i++ {
		fmt.Scan(&p)
		fmt.Scan(&a1)
		fmt.Scan(&a2)
		fmt.Scan(&b1)
		fmt.Scan(&b2)
		for x:=a1; x<b1; x+=1{
			for y:=a2; y<b2; y+=1{
				a[x][y] ^= p
			}
		}
	}

	for i := range a {
        fmt.Printf("%v\n", i)

    }
}

File: HackerEarth/missiles.py
'''
3
3
3 3 1 3 2
2 2 1 2 2
3 1 1 2 3
'''
n = int(input())
field = [[0 for x in range(n)] for y in range(n)]
nom = int(input())
for _ in range(nom):
    p,a1,a2,b1,b2 = [int(x) for x in input().split()]
    i, j = a1-1, a2-1


    for x in range(i,b1):
        for y in range(j, b2):
            field[x][y] ^= p


print(field)

